pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -753,97 +890,156 @@ func (nc *Controller) monitorNodeHealth() error {
-						taintToAdd := *NotReadyTaintTemplate
-						if !nodeutil.SwapNodeControllerTaint(nc.resourcePartitionClient, []*v1.Taint{&taintToAdd}, []*v1.Taint{UnreachableTaintTemplate}, node) {
-							klog.Errorf("Failed to instantly swap UnreachableTaint to NotReadyTaint. Will try again in the next cycle.")
-						}
-					} else if nc.markNodeForTainting(node) {
-						klog.V(2).Infof("Node %v is NotReady as of %v. Adding it to the Taint queue.",
-							node.Name,
-							decisionTimestamp,
-						)
-					}
-				} else {
-					if decisionTimestamp.After(nc.nodeHealthMap[node.Name].readyTransitionTimestamp.Add(nc.podEvictionTimeout)) {
-						if nc.evictPods(node) {
-							klog.V(2).Infof("Node is NotReady. Adding Pods on Node %s to eviction queue: %v is later than %v + %v",
-								node.Name,
-								decisionTimestamp,
-								nc.nodeHealthMap[node.Name].readyTransitionTimestamp,
-								nc.podEvictionTimeout,
-							)
-						}
-					}
+			for _, tpManager := range nc.tenantPartitionManagers {
+				pods, err := tpManager.PodByNodeNameLister(node.Name)
+				if len(pods) == 0 && err == nil {
+					continue
 				}
-			}
-			if observedReadyCondition.Status == v1.ConditionUnknown {
-				if nc.useTaintBasedEvictions {
-					// We want to update the taint straight away if Node is already tainted with the UnreachableTaint
-					if taintutils.TaintExists(node.Spec.Taints, NotReadyTaintTemplate) {
-						taintToAdd := *UnreachableTaintTemplate
-						if !nodeutil.SwapNodeControllerTaint(nc.resourcePartitionClient, []*v1.Taint{&taintToAdd}, []*v1.Taint{NotReadyTaintTemplate}, node) {
-							klog.Errorf("Failed to instantly swap UnreachableTaint to NotReadyTaint. Will try again in the next cycle.")
-						}
-					} else if nc.markNodeForTainting(node) {
-						klog.V(2).Infof("Node %v is unresponsive as of %v. Adding it to the Taint queue.",
-							node.Name,
-							decisionTimestamp,
-						)
-					}
-				} else {
-					if decisionTimestamp.After(nc.nodeHealthMap[node.Name].probeTimestamp.Add(nc.podEvictionTimeout)) {
-						if nc.evictPods(node) {
-							klog.V(2).Infof("Node is unresponsive. Adding Pods on Node %s to eviction queues: %v is later than %v + %v",
-								node.Name,
-								decisionTimestamp,
-								nc.nodeHealthMap[node.Name].readyTransitionTimestamp,
-								nc.podEvictionTimeout-gracePeriod,
-							)
-						}
+
+				if err != nil {
+					utilruntime.HandleError(fmt.Errorf("unable to list pods of node %v: %v", node.Name, err))
+					if currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue {
+						// If error happened during node status transition (Ready -> NotReady)
+						// we need to mark node for retry to force MarkPodsNotReady execution
+						// in the next iteration.
+						nc.nodesToRetry.Store(node.Name, struct{}{})
+						// mark the node for retry and continue for other TPs
+						donePodEvictionOnNode = false
+						continue
```

Comment:
Is "continue" inside or outside of is clause?

Reply:
inside. the goal for this two-loops are ensure retryNode only deleted for pods on both TPs were evicted, and the flow runs for each nodes.

Reply:
Community change has continue outside of if clause.

Reply:
logically they are no difference. but i will move out to align with the community changes.

Reply:
fixed.